### DISCUSSION TITLE
Created a PAT, but git keeps sending a password

### DISCUSSION THREAD

**Discussion by eygilbert** [community/community#23909] [New to GitHub]
<p>I normally use TortoiseGit for GitHub access (using Windows10). Today, after getting messages about passwords no longer being accepted, I created a wincred PAT, cleared the saved authentication data cache from TortoiseGit, then tried to do a push. But GitHub complains that git is still trying to send a password. I’m not really familiar with the git command line commands. I’ve been reading the docs for quite a while now, but not getting anywhere. I installed the latest git version. My global .gitconfig file has these entries:</p>
<p>[credential “<a href="https://github.com" rel="noopener nofollow ugc">https://github.com</a>”]<br>
helper = wincred<br>
username = eygilbert</p>
<p>Can someone help me get git to start using the PAT instead of a password? Thanks.</p>

**Comment by zachPorras** [comment]
<p><a href="https://github.community/t/token-not-working-in-command-line/195566/2">Having a similar issue</a></p>
<p>I’m watching this thread, feel free to watch mine in case I get a solution from someone.  I did get some advice but it did not solve it for me.</p>
<p>Good luck!</p>

**Comment by eygilbert** [comment]
<p>It seems that my password is cached somewhere and git is sending it automatically. I need to delete this, so that I can get it to prompt for credentials and input the PAT that I have created. But I cannot find where the password is cached. I’ve looked in the Windows Credentials Manager but there are no entries there for git. I don’t see a password cached in any of the .git config files. I’m really stuck. How can I purge this cached password? When I installed git I also installed Git Credential Manager Core. Is it possibly cached there? I don’t know to how I can tell.</p>

**Comment by tajmone** [comment]
<aside class="quote no-group" data-username="eygilbert" data-post="3" data-topic="195592">
<div class="title">
<div class="quote-controls"></div>
 eygilbert:</div>
<blockquote>
<p>It seems that my password is cached somewhere and git is sending it automatically.</p>
</blockquote>
</aside>
<p>Yes, that’s indeed the case.</p>
<p><a href="https://git-scm.com/docs/git-credential-store" class="onebox" target="_blank" rel="noopener nofollow ugc">https://git-scm.com/docs/git-credential-store</a></p>
<aside class="quote no-group" data-username="eygilbert" data-post="3" data-topic="195592">
<div class="title">
<div class="quote-controls"></div>
 eygilbert:</div>
<blockquote>
<p>When I installed git I also installed Git Credential Manager Core. Is it possibly cached there? I don’t know to how I can tell.</p>
</blockquote>
</aside>
<p>Possibly, if you have enabled Git to use that instead of its default credentials tool.</p>
<p>A good file to check is <code>.git-credentials</code> in your User folder (talking about Windows here: <code>C:\Users\YourUsername\</code>), and also in the <code>.ssh/</code> subfolder therein, in case you’re using SSH with Git.</p>
<p>If not, you’ll have to query your Git configuration to check which credentials manager it’s using, and where it might be storing them:</p>

**Comment by eygilbert** [comment]
<p>Tajmone, thanks. I don’t find a .git-credentials file anywhere.<br>
I have this entry in .gitconfig</p>
<p>[credential]<br>
helper = manager-core</p>
<p>I queried the gcm-core version:</p>
<p>d:\projects&gt;git credential-manager-core --version<br>
2.0.475+27fbadca1d</p>
<p>After reading the docs about gcm-core, it looks like I need to send the erase command, although I’m not sure, it’s not at all clear to me what I should do. Anyway, I tried this:</p>
<p>d:\projects&gt;git credential-manager-core erase<br>
protocol=https<br>
<a href="http://host=github.com" class="onebox" target="_blank" rel="noopener nofollow ugc">host=github.com</a><br>
username=eygilbert</p>
<p>But it still has the password cached. Can you tell me what the commands should be to erase the password?</p>
<p>The docs for gcm-core say this about the commands get / store / erase:</p>
<p>“Commands for interaction with Git. You shouldn’t need to run these manually.” Which makes me think I’m looking in the wrong place.</p>

**Comment by byrneh** [comment]
<p>On your device type in credential manager on windows search anf launch app. Select the correct tab and delete the entry referring to git.<br>
Next time you perform a Git command that requires authentication you will be prompted to enter credential</p>

**Comment by byrneh** [comment]
<aside class="quote quote-modified" data-post="2" data-topic="163304">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/how-to-start-using-a-personal-access-token-to-authenticate-git-on-windows/163304/2">How to start using a personal access token to authenticate git on windows</a> <a class="badge-wrapper  bullet" href="/c/github-help/how-to-use-git-and-github/21"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #ffd1ac;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Ask and answer questions regarding Git and GitHub.">How to use Git and GitHub</span></a>
  </div>
  <blockquote>
    If you are still using an existing cached password for you login you will need to remove it, the below will help, hopefully slight_smile 
Launch ‘Credential Manager’ on your Windows device. 
Switch to tab Windows Credentials (it default displayed tab Web Credentials on my device). 
A list of locally cached credentials are then shown 
Find an entry starting something like git:htttps//login@github.com (this will be an exist cached entry for your username/password.  You can safely delete this by …
  </blockquote>
</aside>

**Comment by eygilbert** [comment]
<p>Thank you byrneh. I have looked at all the Windows credentials listed in Credential Manager, and none of them are anything related to git.</p>

**Comment by byrneh** [comment]
<p><a class="mention" href="/eygilbert">@eygilbert</a> are you using git command line or Tortoise Git?<br>
If using git command line that is installed with git-scm install. It will will only use the Git credential manager specified in you git config for caching, you can even disable the use of crendential manager if you so choose.<br>
If you type <em>git version</em> you should see something like git version 2.32.0.windows.1 returned.<br>
To see in detail you local git configuration type<br>
<em>git config --list --show-origin --show-scope</em></p>
<p>I don’t use Tortoise Git but I am told this suggestion may work.<br>
Go to your TortoiseGit settings for your repository, then go to the local .git/config… It should open in Notepad.<br>
After that, edit the URL to the repository, in the format of github username:access_token@rest/of/the/url.git</p>

**Comment by eygilbert** [comment]
<blockquote>
<p>are you using git command line or Tortoise Git?</p>
</blockquote>
<p>I normally use TortoiseGit, but since my problem seems to be with the underlying command-line git, I have been working with that to try to solve the problem. Installing TortoiseGit requires that I also install the command-line git.</p>
<blockquote>
<p>If you type <em>git version</em> you should see something like git version 2.32.0.windows.1 returned.</p>
</blockquote>
<p>d:\projects&gt;git version<br>
git version 2.32.0.windows.2</p>
<p>d:\projects&gt;git config --list --show-origin<br>
file:C:/Program Files/Git/etc/gitconfig credential.helper=manager-core<br>
file:C:/Program Files/Git/etc/gitconfig diff.astextplain.textconv=astextplain<br>
file:C:/Program Files/Git/etc/gitconfig filter.lfs.clean=git-lfs clean – %f<br>
file:C:/Program Files/Git/etc/gitconfig filter.lfs.smudge=git-lfs smudge – %f<br>
file:C:/Program Files/Git/etc/gitconfig filter.lfs.process=git-lfs filter-process<br>
file:C:/Program Files/Git/etc/gitconfig filter.lfs.required=true<br>
file:C:/Program Files/Git/etc/gitconfig http.sslbackend=openssl<br>
file:C:/Program Files/Git/etc/gitconfig http.sslcainfo=C:/Program Files/Git/mingw64/ssl/certs/ca-bundle.crt<br>
file:C:/Program Files/Git/etc/gitconfig core.autocrlf=input<br>
file:C:/Program Files/Git/etc/gitconfig core.fscache=true<br>
file:C:/Program Files/Git/etc/gitconfig core.symlinks=false<br>
file:C:/Program Files/Git/etc/gitconfig core.editor=notepad<br>
file:C:/Program Files/Git/etc/gitconfig pull.rebase=false<br>
file:C:/Program Files/Git/etc/gitconfig credential.<a href="https://dev.azure.com.usehttppath=true" rel="noopener nofollow ugc">https://dev.azure.com.usehttppath=true</a><br>
file:C:/Program Files/Git/etc/gitconfig init.defaultbranch=master<br>
file:c:/Users/family/Documents/.gitconfig       user.name=Ed Gilbert<br>
file:c:/Users/family/Documents/.gitconfig       <a href="mailto:user.email=eygilbert@gmail.com">user.email=eygilbert@gmail.com</a><br>
file:c:/Users/family/Documents/.gitconfig       credential.<a href="https://github.com.helper=wincred" rel="noopener nofollow ugc">https://github.com.helper=wincred</a><br>
file:c:/Users/family/Documents/.gitconfig       credential.<a href="https://github.com.username=eygilbert" rel="noopener nofollow ugc">https://github.com.username=eygilbert</a><br>
file:c:/Users/family/Documents/.gitconfig       gui.recentrepo=D:/Projects/Git/GuiCheckers</p>
<blockquote>
<p>After that, edit the URL to the repository, in the format of github username:access_token@rest/of/the/url.git</p>
</blockquote>
<p>Here is the entry from the config file of one of my repos:</p>
<pre><code class="lang-auto">[remote "origin"]
	url = https://github.com/eygilbert/egdb_intl
	fetch = +refs/heads/*:refs/remotes/origin/*
</code></pre>
<p>It’s not in the format that you gave. I’m not sure how to proceed from there. Should I simply replace my username with username:access_token in the line below?</p>
<pre><code>url = https://github.com/eygilbert/egdb_intl</code></pre>

**Comment by eygilbert** [comment]
<p>Ah, I misunderstood your suggestion. I edited the config file to look like this:</p>
<p><code>url = https://token:ghp_........@github.com/eygilbert/egdb_intl</code></p>
<p>and now I am able to push to this repos. Thanks for the help!</p>

**Comment by eygilbert** [comment, marked as answer]
<p>I finally found the root cause of my problem. The github username and password were cached in a _netrc file in my Documents directory. What’s really insidious is that none of the git commands seemed to be able to either display or delete this cache. With this problem fixed, git is now using the PAT stored in Windows Credential manager, and I don’t have to edit the remote origin url in every config file of every local repository to make authentication work.</p>

**Comment by tajmone** [comment]
<aside class="quote no-group" data-username="eygilbert" data-post="12" data-topic="195592">
<div class="title">
<div class="quote-controls"></div>
 eygilbert:</div>
<blockquote>
<p>I finally found the root cause of my problem.</p>
</blockquote>
</aside>
<p>Excellent! Mark your solution as the Solution, so other users can quickly find it and benefit from it.</p>

**Comment by Naheulf** [comment]
<p>Thanks, it’s that. However I have <strong>two</strong> entries for Github… Grrr</p>

**Comment by tajmone** [comment]
<p>Hi <a class="mention" href="/naheulf">@Naheulf</a>! Nice to see you here on the Community Forum! (I’ve just labeled your PR for the French docs, thanks!)</p>

**Comment by penguin020** [comment]
I had a similar problem.  It turned out to be I had 2 entries in the windows credential manager: one for https://myusername@xxx,gitlab.com, the other just for https://xxx,gitlab.com, or I think that was the reason.  deleting one seems to have fixed the issue, but who knows?
